client/client.go

```diff
@@ -134,6 +138,58 @@ func (c *Client) do(req *fasthttp.Request, resp *fasthttp.Response, readTimeout,
 	return nil
 }
 
+func (c *Client) cachedTLSConfig(addr string) *tls.Config {
+	c.tlsConfigMapLock.Lock()
+	if c.tlsConfigMap == nil {
```

Comment:
We have a function called `NewClient`. Can we move initialization there?

Reply:
done